include/cantera/zeroD/WallFactory.h

```diff
@@ -0,0 +1,61 @@
+//! @file WallFactory.h
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#ifndef WALL_FACTORY_H
+#define WALL_FACTORY_H
+
+#include "cantera/base/FactoryBase.h"
+#include "cantera/zeroD/Wall.h"
+
+namespace Cantera
+{
+
+class WallFactory : public Factory<WallBase>
+{
+public:
+    static WallFactory* factory() {
+        std::unique_lock<std::mutex> lock(wall_mutex);
+        if (!s_factory) {
+            s_factory = new WallFactory;
+        }
+        return s_factory;
+    }
+
+    virtual void deleteFactory() {
+        std::unique_lock<std::mutex> lock(wall_mutex);
+        delete s_factory;
+        s_factory = 0;
+    }
+
+    /**
+     * Create a new Wall.
+     * @param n the type to be created.
+     */
+    virtual WallBase* newWall(int n);
+    virtual WallBase* newWall(const std::string& wallType);
```

Comment:
I would prefer to have only the string-based constructor.

Reply:
At this point, `ctreactor.cpp` still depends on integer-based constructors; I kept this approach to mimic implementations of `Reactor`-type objects.

Reply:
As mentioned above, this involves major changes in CLIB - which builds on the constructor int wall_new(int type), - potentially breaking backwards compatibility. I propose to defer elimination/deprecation of magic numbers in a separate effort.